Deceive/Utils.cs

```diff
@@ -268,8 +273,64 @@ public static string GetRiotClientPath()
                     return (string)entry.Value;
                 }
             }
+            return null;
+        }
+
+        //class for storing LCU API port and auth token
+        public class LCUAPIPortToken
+        {
+            public LCUAPIPortToken(string port, string token)
+            {
+                Port = port;
+                Token = token;
+            }
+
+            public string Port { get; }
 
+            public string Token { get; }
+        }
+
+        //get LCU API port and auth token from commandline options
+        public static LCUAPIPortToken GetAPIPortAndToken(Process process)
+        {
+            using (ManagementObjectSearcher searcher = new ManagementObjectSearcher("SELECT CommandLine FROM Win32_Process WHERE ProcessId = " + process.Id))
+            using (ManagementObjectCollection objects = searcher.Get())
+            {
+                var commandLine = (string)objects.Cast<ManagementBaseObject>().SingleOrDefault()["CommandLine"];
+                try
+                {
+                    var port = PORT_REGEX.Match(commandLine).Groups[1].Value;
+                    var token = AUTH_TOKEN_REGEX.Match(commandLine).Groups[1].Value;
+                    return new LCUAPIPortToken(port, token);
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine(ex.Message);
+                }
+            }
             return null;
         }
+
+        /*
+         * sends our masked availability to LCU for display to local player, instead of showing normal status
+         * LCU will only display availability, so still shows 'Creating Normal Game' or 'In Game',
+         * but only locally, since Deceive masks whole presence with 'gameStatus' as 'outOfGame'.
+         * Passing this/whole presence to LCU doesn't make a difference as LCU just overrides it.
+         */
+        public static void SendStatusToLCU(string status)
+        {
+            foreach (Process process in Process.GetProcessesByName("LeagueClientUx"))
+            {
+                var portToken = GetAPIPortAndToken(process);
```

Comment:
Should you handle `portToken == null` here? I doubt there's any situations in which this can happen (especially now that we check for admin), but better safe than sorry.

Reply:
Yeah, I was pretty sure it would never happen, but can definitely check for that.